### ISSUE TITLE
transitiveRemediation: vulnerable version can get added back by `npm install`

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#8941]
**What Renovate type, platform and version are you using?**

**Describe the bug**

A transitive remediation PR may still contain the vulnerable dependency version added back as a sub-dependency.

Example: https://github.com/renovate-tests/hubot-cne/pull/3/files (see line 1697)

**Relevant debug logs**

```
{"dependencies":["base64-url"],"msg":"processBranch with 1 upgrades"}
{"msg":"Setting current branch to master"}
{"msg":"getBranchPr(renovate/npm-base64-url-vulnerability)"}
{"msg":"findPr(renovate/npm-base64-url-vulnerability, undefined, open)"}
{"msg":"branchExists=false"}
{"msg":"Branch has 1 upgrade(s)"}
{"msg":"recreateClosed is false"}
{"msg":"findPr(renovate/npm-base64-url-vulnerability, chore(deps): update dependency base64-url to 2.0.0 [security], !open)"}
{"msg":"prAlreadyExisted=false"}
{"msg":"Checking schedule(, null)"}
{"msg":"No schedule defined"}
{"msg":"Branch needs creating"}
{"msg":"Using reuseExistingBranch: false"}
{"msg":"manager.getUpdatedPackageFiles() reuseExistinbranch=false"}
{"msg":"npm.updateLockedDependency: base64-url@1.2.1 -> 2.0.0 [package-lock.json]"}
{"msg":"Found matching dependencies with length 1"}
{"msg":"Finding first version of uid-safe starting with 2.0.0 which supports >= base64-url@2.0.0"}
{"msg":"base64-url has been removed from uid-safe@2.1.4"}
{"msg":"Update of base64-url to 2.0.0 can be achieved due to parent uid-safe"}
{"msg":"npm.updateLockedDependency: uid-safe@2.0.0 -> 2.1.4 [package-lock.json]"}
{"msg":"Found matching dependencies with length 1"}
{"msg":"Finding first version of express-session starting with 1.11.3 which supports >= uid-safe@2.1.4"}
{"msg":"uid-safe needs express-session@1.14.0 which uses constraint \"~2.1.1\" in order to update to 2.1.4"}
{"msg":"Update of uid-safe to 2.1.4 can be achieved due to parent express-session"}
{"msg":"npm.updateLockedDependency: express-session@1.11.3 -> 1.14.0 [package-lock.json]"}
{"msg":"Found matching dependencies with length 1"}
{"msg":"Finding first version of connect starting with 2.30.2 which supports >= express-session@1.14.0"}
{"msg":"express-session has been removed from connect@3.0.0"}
{"msg":"Update of express-session to 1.14.0 can be achieved due to parent connect"}
{"msg":"npm.updateLockedDependency: connect@2.30.2 -> 3.0.0 [package-lock.json]"}
{"msg":"Found matching dependencies with length 1"}
{"msg":"Finding first version of express starting with 3.21.2 which supports >= connect@3.0.0"}
{"msg":"connect has been removed from express@4.0.0"}
{"msg":"Update of connect to 3.0.0 can be achieved due to parent express"}
{"msg":"npm.updateLockedDependency: express@3.21.2 -> 4.0.0 [package-lock.json]"}
{"msg":"Found matching dependencies with length 1"}
{"msg":"Finding first version of hubot starting with 3.0.1 which supports >= express@4.0.0"}
{"msg":"express needs hubot@3.1.0 which uses constraint \"^4.16.3\" in order to update to greater than 4.0.0"}
{"msg":"Update of express to 4.0.0 can be achieved due to parent hubot"}
{"msg":"npm.updateLockedDependency: hubot@3.0.1 -> 3.1.0 [package-lock.json]"}
{"msg":"Found matching dependencies with length 1"}
{"msg":"Finding first version of hubot-test-helper starting with 1.9.0 which supports >= hubot@3.1.0"}
{"msg":"hubot needs hubot-test-helper@1.9.0 which uses constraint \">=3.0.0 <10 || 0.0.0-development\" in order to update to 3.1.0"}
{"msg":"Update of hubot to 3.1.0 already achieved in parent hubot-test-helper@1.9.0"}
{"msg":"Updated 1 package files"}
{"msg":"Getting updated lock files"}
{"packageFiles":["package.json"],"msg":"Writing package.json files"}
{"msg":"Writing package-lock.json"}
{"msg":"Writing any updated package files"}
{"msg":"Writing package-lock file: package-lock.json"}
{"msg":"No npmrc file found in repository"}
{"msg":"Writing updated .npmrc file to /tmp/renovate/repos/github/renovate-tests/hubot-cne/.npmrc"}
{"msg":"Generating package-lock.json for ."}
{"msg":"Spawning npm install to create /tmp/renovate/repos/github/renovate-tests/hubot-cne/package-lock.json"}
{"msg":"Updating lock file only"}
{"msg":"Using node constraint \"8\" from .nvmrc"}
{"msg":"Using docker to execute"}
{"depName":"docker.io/renovate/node","scheme":"npm","constraint":"8","msg":"Found version constraint - checking for a compatible image to use"}
{"depName":"docker.io/renovate/node","scheme":"npm","constraint":"8","version":"8.17.0","msg":"Found compatible image version"}
{"image":"docker.io/renovate/node","tagConstraint":"8","tagVersioning":"npm","tag":"8.17.0","msg":"Resolved tag constraint"}
{"msg":"Docker image is already prefetched: docker.io/renovate/node:8.17.0"}
{"command":"docker run --rm --name=renovate_node --label=renovate_child -v \"/tmp/renovate/repos/github/renovate-tests/hubot-cne\":\"/tmp/renovate/repos/github/renovate-tests/hubot-cne\" -v \"/tmp/renovate/cache\":\"/tmp/renovate/cache\" -e NPM_CONFIG_CACHE -e npm_config_store -w \"/tmp/renovate/repos/github/renovate-tests/hubot-cne\" docker.io/renovate/node:8.17.0 bash -l -c \"npm i -g 'npm@<7' && hash -d npm && npm install --package-lock-only --ignore-scripts --no-audit\"","msg":"Executing command"}
{"cmd":"docker run --rm --name=renovate_node --label=renovate_child -v \"/tmp/renovate/repos/github/renovate-tests/hubot-cne\":\"/tmp/renovate/repos/github/renovate-tests/hubot-cne\" -v \"/tmp/renovate/cache\":\"/tmp/renovate/cache\" -e NPM_CONFIG_CACHE -e npm_config_store -w \"/tmp/renovate/repos/github/renovate-tests/hubot-cne\" docker.io/renovate/node:8.17.0 bash -l -c \"npm i -g 'npm@<7' && hash -d npm && npm install --package-lock-only --ignore-scripts --no-audit\"","durationMs":23429,"stdout":"/home/ubuntu/.npm-global/bin/npm -> /home/ubuntu/.npm-global/lib/node_modules/npm/bin/npm-cli.js\n/home/ubuntu/.npm-global/bin/npx -> /home/ubuntu/.npm-global/lib/node_modules/npm/bin/npx-cli.js\n+ npm@6.14.11\nadded 435 packages from 889 contributors in 7.481s\nadded 468 packages in 13.664s\n","stderr":"npm WARN deprecated coffee-script@1.6.3: CoffeeScript on NPM has moved to \"coffeescript\" (no hyphen)\nnpm WARN deprecated connect@2.30.2: connect 2.x series is deprecated\n","msg":"exec completed"}
{"msg":"package-lock.json needs updating"}
{"updatedArtifacts":["package-lock.json"],"msg":"Updated 1 lock files"}
{"msg":"1 file(s) to commit"}
{"msg":"Committing files to branch renovate/npm-base64-url-vulnerability"}
{"name":"renovate","hostname":"mbp16.lan","pid":34119,"level":30,"logContext":"OijxQft3M","repository":"renovate-tests/hubot-cne","commitSha":"d54f468","msg":"Branch created"}
{"msg":"Ensuring PR"}
{"msg":"There are 0 errors and 0 warnings"}
{"msg":"getBranchPr(renovate/npm-base64-url-vulnerability)"}
{"msg":"findPr(renovate/npm-base64-url-vulnerability, undefined, open)"}
{"prTitle":"chore(deps): update dependency base64-url to 2.0.0 [security]","msg":"Creating PR"}
{"title":"chore(deps): update dependency base64-url to 2.0.0 [security]","head":"renovate-tests:renovate/npm-base64-url-vulnerability","base":"master","draft":false,"msg":"Creating PR"}
{"pr":3,"draft":false,"msg":"PR created"}
{"msg":"Adding labels '' to #3"}
{"name":"renovate","hostname":"mbp16.lan","pid":34119,"level":30,"logContext":"OijxQft3M","repository":"renovate-tests/hubot-cne","pr":3,"prTitle":"chore(deps): update dependency base64-url to 2.0.0 [security]","msg":"PR created"}
{"msg":"Created Pull Request #3"}
{"query":"\n      query {\n        repository(owner: \"renovate-tests\", name: \"hubot-cne\") {\n          pullRequests(states: [CLOSED, MERGED], orderBy: {field: UPDATED_AT, direction: DESC}) {\n            pageInfo {\n              endCursor\n              hasNextPage\n            }\n            nodes {\n              number\n              state\n              headRefName\n              title\n              comments(last: 100) {\n                nodes {\n                  databaseId\n                  body\n                }\n              }\n            }\n          }\n        }\n      }\n      ","msg":"No graphql data, returning empty list"}
{"msg":"Getting comments for #3"}
{"msg":"Found 0 comments"}
{"automerge":false,"automergeType":"pr","automergeComment":"automergeComment","msg":"Checking #3 for automerge"}
{"msg":"No automerge"}
```

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/master/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] This is a really small bug, it does not need a reproduction (think small typo)
- [ ] I have provided a minimal reproduction repository
- [x] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

**Comment by rarkins** [comment]
Probably necessary: run `npm` as part of `updateLockedDependency()` so that the result can be validated afterwards.

Maybe possible: if the vulnerable dependency comes *back* then try again with the new package/lock file

**Comment by rarkins** [comment]
The above example was fixed as a bug was found which caused it, but there's a new edge case here: https://github.com/LastKing/markdown-menu/pull/32/files

In short, the vulnerability fix release has been badly versions as `5.0.0-security.0`, which is (a) unstable and (b) *less* than the `5.0.0` release it's fixing. Although Renovate sets the version, `npm install` reverts it.

Ideally we should abort this PR.
